### DISCUSSION TITLE
How to stop/update Caching vscode.MarkdownString?

### DISCUSSION THREAD

**Discussion by PaulleDemon** [microsoft/vscode-discussions#1279] [Extension Development QnA] [answered]
I am developing an extension and I am using an Image in MarkdownString that's being cached
```js
new vscode.MarkdownString(`<img src='${imgUri.path}'  width='200' height='200'/>`);
```

I am updating existing image without creating a new Image, for this reason, imgUri Path remains same. The problem is that the image is not being updated when `MarkdownString` is being displayed. 

I even tried cache busting using query params
```js
new vscode.MarkdownString(`<img src='${imgUri.path}?time=${new Date().getTime()}' alt="rendered preview" />`);
```
but it still doesn't update immediately. Is there a way to update the cache?

**Comment by isidorn** [comment]
I do not know. Maybe @mjbvz could help

**Comment by mjbvz** [comment]
Where is the markdown being rendered?

**Reply by PaulleDemon** [reply, marked as answer]
Hi sorry for the late reply, I was't notified. 

The markdown is being rendered in **Hover**. Right now, I decided to create a temp file to update the preview and then delete that temp file. You can check it out here: https://github.com/PaulleDemon/Hover-Preview/blob/fbd8b6d05d9a4b466827a015e788e1bf7492ec2c/src/extension.js#L92
